Contrib/ChiralPairs/ChiralDescriptors.py

```diff
@@ -0,0 +1,300 @@
+            bonds.append(b.GetIdx())
+    return bonds
+
+def getAromaticBondsSubstituent(mol, subAtoms):
+    bonds = getBondsSubstituent(mol, subAtoms)
+    aroBonds=0
+    for b in bonds:
+        if mol.GetBondWithIdx(b).GetIsAromatic():
+            aroBonds+=1
+    return aroBonds
+
+def getRotatableBondsSubstituent(mol, subAtoms):
+    rotatableBond = Chem.MolFromSmarts('[!$(*#*)&!D1]-&!@[!$(*#*)&!D1]')
+    matches = mol.GetSubstructMatches(rotatableBond)
+    numRotBonds=0
+    for a1,a2 in matches:
+        if a1 in subAtoms and a2 in subAtoms:
+            numRotBonds+=1
+    return numRotBonds
+
+
+substituentDescriptor = namedtuple('substituentDescriptor',
+                                   ['size','relSize','numNO','relNumNO','relNumNO_2','pathLength','relPathLength','relPathLength_2',
+                                    'sharedNeighbors', 'numRotBonds', 'numAroBonds'])    
+
+def calcSizeSubstituents(mol, sub_dict, sharedNeighbors_dict, maxShell_dict):  
+    sizeDict=defaultdict()
+    numAtoms = mol.GetNumAtoms()
+    #print(sharedNeighbors_dict)
+    for sidx, sub in sorted(sub_dict.items(), key= lambda x: len(x[1])):
+        size = _getSizeOfSubstituents(sub, sharedNeighbors_dict)
+        numNOs=0
+        numShared=0
+        # determine the number of oxygen and nitrogen atoms
+        for i in sub:
+            if mol.GetAtomWithIdx(i).GetAtomicNum() in [7,8]:
+                numNOs+=1.0/sharedNeighbors_dict[i]
+            if sharedNeighbors_dict[i] > 1:
+                numShared+=1
+        numRotBs = getRotatableBondsSubstituent(mol, set(sub))
+        aroBonds = getAromaticBondsSubstituent(mol, set(sub))
+        # fill the substituentDescriptor tuple
+        sizeDict[sidx]=substituentDescriptor(size=size,relSize=size/numAtoms,numNO=numNOs,relNumNO=numNOs/numAtoms,
+                                             relNumNO_2=numNOs/size,pathLength=maxShell_dict[sidx],
+                                             relPathLength=maxShell_dict[sidx]/numAtoms,relPathLength_2=maxShell_dict[sidx]/size,
+                                                 sharedNeighbors=numShared, numRotBonds=numRotBs, numAroBonds=aroBonds)
+    # if we have less then 4 substituents the missing ones need to be an hydrogen atoms
+    if len(sizeDict) < 4:
+        for i in range(4-len(sizeDict)):
+            sizeDict['H'+str(i)]=substituentDescriptor(size=0,relSize=0,numNO=0,relNumNO=0,relNumNO_2=0,
+                                                       pathLength=0,relPathLength=0,relPathLength_2=0,sharedNeighbors=0, numRotBonds=0,
+                                                       numAroBonds=0)
+    return sizeDict
+
+
+# Visualization of the substituents
+def visualizeSubstituentsGrid(mol, aIdx, molSize=(300,150), kekulize=True,):
+    dists = Chem.GetDistanceMatrix(mol)
+    idxChiral = Chem.FindMolChiralCenters(mol)[0][0]
+    subDict, sn_dict, maxShell_dict = determineAtomSubstituents(aIdx, mol, dists, False)
```

Comment:
you could be more consistent with your variable names. Here is a mixture of camelcase and snakecase. 

A more general comment, I'm not a big fan of giving the basic types like `_dict` at suffix of names. Having read a lot of code, this just increases the text without helping. The type is usually clear from the context.